src/librustc_incremental/persist/fs.rs

```diff
@@ -234,10 +234,19 @@ pub fn prepare_session_directory(tcx: TyCtxt) -> Result<bool, ()> {
         let print_file_copy_stats = tcx.sess.opts.debugging_opts.incremental_info;
 
         // Try copying over all files from the source directory
-        if copy_files(&session_dir, &source_directory, print_file_copy_stats).is_ok() {
+        if let Ok(allows_links) = copy_files(&session_dir, &source_directory,
+                                             print_file_copy_stats) {
             debug!("successfully copied data from: {}",
                    source_directory.display());
 
+            if !allows_links {
+                tcx.sess.warn(&format!("Hard linking files in the incremental compilation
+                    cache failed. Copying files instead. Consider moving the cache directory to a
+                    file system which supports hard linking in session dir `{}`"
```

Comment:
If you have a string that spans multiple lines, you can place a `\` at the end of the line. That will (iirc) cause the parser to ignore the leading whitespace in the next line:
```rust
// This...
let s = "123
         456"
// ... is equivalent to this:
let s = "123\n         456"
// while this:
let s = "123\
         456"
// is equivalent to this:
let s = "123456"
```